### PR TITLE
add single file STAC extension

### PR DESCRIPTION
**Pull request by matthewhanson**
**Related Issue(s):** #385 

**Proposed Changes:**

1. Implement a format for a single file STAC that combines Collections, Items (and optionally search parameters) in a single file

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: extensions/README.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ An extension can add new fields to STAC entities (content extension), or can add
 | [Point Cloud](pointcloud/README.md) (`pc`)                   | Item             | Provides a way to describe point cloud datasets. The point clouds can come from either active or passive sensors, and data is frequently acquired using tools such as LiDAR or coincidence-matched imagery. | *Proposal* |
 | [SAR](sar/README.md) (`sar`)                                 | Item             | Covers synthetic-aperture radar data that represents a snapshot of the earth for a single date and time. | *Proposal* |
 | [Scientific](scientific/README.md) (`sci`)                   | Item             | Scientific metadata is considered to be data that indicate from which publication data originates and how the data itself should be cited or referenced. | *Proposal* |
+| [Single File STAC](single-file-stac/README.md) (-)                   | Catloag +Collection +Item | An extension to provide a set of Collections and Items as a single file catalog. | *Proposal* |
```
Comment (cholmes) [inline extensions/README.md]: Catloag -> Catalog

**Comment by m-mohr** [comment]
Maybe a question to answer in the README: Do the common properties (in collection `properties`) stay in there or are these already merged into the Item `properties`?

File: extensions/README.md [hunk 2]
```diff
@@ -54,8 +54,12 @@ An extension can add new fields to STAC entities (content extension), or can add
 | [Label](label/README.md) (`label`)                           | Item             | Items that relate labeled AOIs with source imagery | *Proposal* |
 | [Point Cloud](pointcloud/README.md) (`pc`)                   | Item             | Provides a way to describe point cloud datasets. The point clouds can come from either active or passive sensors, and data is frequently acquired using tools such as LiDAR or coincidence-matched imagery. | *Proposal* |
 | [SAR](sar/README.md) (`sar`)                                 | Item             | Covers synthetic-aperture radar data that represents a snapshot of the earth for a single date and time. | *Proposal* |
+<<<<<<< HEAD
```
Comment (m-mohr) [inline extensions/README.md]: A merge left-over.

File: extensions/README.md [hunk 3]
```diff
@@ -54,8 +54,12 @@ An extension can add new fields to STAC entities (content extension), or can add
...
 | [Scientific](scientific/README.md) (`sci`)                   | Item             | Scientific metadata is considered to be data that indicate from which publication data originates and how the data itself should be cited or referenced. | *Proposal* |
 | [Single File STAC](single-file-stac/README.md) (-)                   | Catloag +Collection +Item | An extension to provide a set of Collections and Items as a single file catalog. | *Proposal* |
+=======
+| [Scientific](scientific/README.md) (`sci`)                   | Item +Collection | Scientific metadata is considered to be data that indicate from which publication data originates and how the data itself should be cited or referenced. | *Proposal* |
+>>>>>>> a76d904270c4a369582b2319daa655e97d3f1b02
```
Comment (m-mohr) [inline extensions/README.md]: Merge left-over

File: extensions/README.md [hunk 4]
```diff
@@ -54,12 +54,8 @@ An extension can add new fields to STAC entities (content extension), or can add
...
 | [SAR](sar/README.md) (`sar`)                                 | Item             | Covers synthetic-aperture radar data that represents a snapshot of the earth for a single date and time. | *Proposal* |
-<<<<<<< HEAD
-| [Scientific](scientific/README.md) (`sci`)                   | Item             | Scientific metadata is considered to be data that indicate from which publication data originates and how the data itself should be cited or referenced. | *Proposal* |
-| [Single File STAC](single-file-stac/README.md) (-)                   | Catloag +Collection +Item | An extension to provide a set of Collections and Items as a single file catalog. | *Proposal* |
-=======
 | [Scientific](scientific/README.md) (`sci`)                   | Item +Collection | Scientific metadata is considered to be data that indicate from which publication data originates and how the data itself should be cited or referenced. | *Proposal* |
->>>>>>> a76d904270c4a369582b2319daa655e97d3f1b02
+| [Single File STAC](single-file-stac/README.md) (-)           | Catalog +Collection +Item | An extension to provide a set of Collections and Items as a single file catalog. | *Proposal* |
```
Comment (m-mohr) [inline extensions/README.md]: Is this actually an extension to all of Catalog +Collection +Item? Isn't it more an extension to ItemCollection?

File: CHANGELOG.md [hunk 5]
```diff
@@ -27,6 +27,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - **stac_extensions**: Introduced this field for Items, Catalogs and Collections.
 - Property `summaries` have been added to catalogs and collections.
 - API Transaction extension supports optimistic locking through use of the ETag header.
+- [Single File Catalog extension](extensions/single-file-catalog/README.md) added as a format to have a set of Collections and Items in a single file.
```
Comment (m-mohr) [inline CHANGELOG.md]: The path is incorrect, should be single-file-stac.

**Pull request merged**